src/components/HistoryTable/utils.ts

```diff
@@ -31,24 +40,38 @@ export const processCommits = (
           type: rowType.COMMIT,
           commit: version,
           date: version.createTime,
+          selected: version.order === selectedCommitOrder,
           rowHeight: COMMIT_HEIGHT,
         });
+
+        if (version.order === selectedCommitOrder) {
+          selectedCommitRowIndex = processedCommits.length - 1;
+        }
```

Comment:
Since this code is repeated on line 64 I believe you can pull it out to the top level if statement `if (commitType === rowType.COMMIT)`. 

Tbh you might even be able to move the whole thing up with something like this:

<img width="602" alt="Screen Shot 2022-03-29 at 12 19 50 PM" src="https://user-images.githubusercontent.com/47064971/160658833-9b841360-b6bb-4536-9ad3-2c449eb5c3bb.png">

Reply:
Swapped it out with switch case statement which looks more readable imo lmk what you think.